pyvo/extensions/cadcauth.py

```diff
@@ -0,0 +1,29 @@
+import logging
+import os
+
+import requests
+import requests.auth
+
+class CADCAuth(requests.auth.AuthBase):
```

Comment:
I think `CookieAuth` can be a generic authenticator for when the user already has the cookie. These CADCAuth is in fact a user/password authenticator, which in this case defaults to `CookieAuth` but in other cases can implement `BasicAA` as it was the case not that long ago.

Reply:
It's true, but the hard part isn't the cookie per se, but it's also how to get it, which is equally important to a user.  I don't know where these specific provider auth recipes belong, but I think this logic does belong somewhere (maybe in astroquery?) to assist people that know they have an account and want to use it.  Once we have a generic way to describe where to get the cookie, I think this can get down to one implementation quite easily.

I'm going to try to stick in general to providing some of the SSO approved methods though, like cookies, certs, and maybe open id. 

Doing BasicAA can easily be done by adding another provider that looks like this but doesn't get the cookie.  You could easily plug them into the same authsession too to use them each when they are required